code/BTCUtil.c

```diff
@@ -780,6 +780,17 @@ Datas ckdIndicesFromPath(const char *cStr)
     return DTPopDatas(result);
 }
 
+Data privKeyToHdWallet(Data privKey, const char *passphrase)
+{
+    Data seedPhrase = DataCopy("Bitcoin seed", strlen("Bitcoin seed"));
+
+    Data seed = PBKDF2(toMnemonic(privKey).bytes, passphrase);
+
+    Data hash = hmacSha512(seedPhrase, seed);
+
+    return hdWalletPriv(DataCopyDataPart(hash, 0, 32), DataCopyDataPart(hash, 32, 32));
```

Comment:
To "unindent" the ARC scope by one, this needs to become `return DTPop(...);` where ... is your hdWalletPriv code.